utility/session.go

```diff
@@ -1,157 +1,280 @@
+		var chain string
+		for _, chain = range fisherman.Chains {
+			if chain != s.session.RelayChain {
+				chain = ""
+				continue
+			}
+		}
+		if chain != "" {
+			candidateFishermen = append(candidateFishermen, fisherman)
+		}
+	}
+
+	s.session.Fishermen = pseudoRandomSelection(candidateFishermen, numFishermen, s.sessionIdBz)
+	return nil
 }
 
-// uses the current 'world state' to determine the fishermen in the session
-// 1) get an ordered list of the public keys of fishermen who are:
-//   - actively staked
-//   - staked within geo-zone  (or closest geo-zones)
-//   - staked for relay-chain
-//
-// 2) calls `pseudoRandomSelection(fishermen, numberOfFishPerSession)`
-func (s *session) selectSessionFishermen(numFishermen int) ([]*coreTypes.Actor, types.Error) {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil, nil
+// pseudoRandomSelection returns a random subset of the candidates.
+// TECHDEBT: We are using a `Go` native implementation for a pseudo-random number generator. In order
+// for it to be language agnostic, a general purpose algorithm needs ot be used.
+func pseudoRandomSelection(candidates []*coreTypes.Actor, numTarget int, sessionId []byte) []*coreTypes.Actor {
+	// If there aren't enough candidates, return all of them
+	if numTarget > len(candidates) {
+		logger.Global.Warn().Msgf("pseudoRandomSelection: numTarget (%d) is greater than the number of candidates (%d)", numTarget, len(candidates))
+		return candidates
+	}
+
+	// Take the first 8 bytes of sessionId to use as the seed
+	seed := int64(binary.BigEndian.Uint64(crypto.SHA3Hash(sessionId)[:8]))
+
+	// Retrieve the indices for the candidates
+	actors := make([]*coreTypes.Actor, 0)
+	uniqueIndices := uniqueRandomIndices(seed, int64(len(candidates)), int64(numTarget))
+	for idx := range uniqueIndices {
+		actors = append(actors, candidates[idx])
+	}
+
+	return actors
 }
 
-// 1) passed an ordered list of the public keys of actors and number of nodes
-// 2) pseudo-insert the session `key` string into the list and find the first actor directly below
-// 3) newKey = Hash( key + actor1PublicKey )
-// 4) repeat steps 2 and 3 until all N actor are found
-// FAQ:
-// Q) why do we hash to find a newKey between every actor selection?
-// A) pseudo-random selection only works if each iteration is re-randomized
-//
-//	or it would be subject to lexicographical proximity bias attacks
-//
-//nolint:unused // This is a demonstratable function
-func (s *session) pseudoRandomSelection(orderedListOfPublicKeys []string, numActorsToSelect int) []*coreTypes.Actor {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil
+// OPTIMIZE: Postgres uses a `Twisted Mersenne Twister (TMT)` randomness algorithm.
+// We could potentially look into changing everything into a single SQL query but
+// would nee dto verify that it can be implemented in a platform agnostic way.
```

Comment:
:100: 

I think @dylanlott was pointing out the typo:
```suggestion
// OPTIMIZE: Postgres uses a `Twisted Mersenne Twister (TMT)` randomness algorithm.
// We could potentially look into changing everything into a single SQL query but
// would need to verify that it can be implemented in a platform agnostic way.
```